content/docs/2.6/scalers/datadog.md

```diff
@@ -0,0 +1,101 @@
++++
+title = "Datadog"
+layout = "scaler"
+availability = "v2.6+"
+maintainer = "Community"
+description = "Scale applications based on Datadog."
+go_file = "datadog_scaler"
++++
+
+### Trigger Specification
+
+This specification describes the `datadog` trigger that scales based on a Datadog metric.
+
+```yaml
+triggers:
+- type: datadog
+  metadata:
+    # Required: datadog metric query
+    query: "sum:trace.redis.command.hits{env:none,service:redis}.as_count()"
+    # Required: according to the number of query result, to scale the TargetRef
+    queryValue: "7"
+    # Optional: (Global or Average). Whether the target value is global or average per pod. Default: Average
+    type: "Global"
+    # Optional: The generated metric name would be datadog-redis-hits
+    metricName: "redis-hits"
+    # Optional: The time window (in seconds) to retrieve metrics from Datadog. Default: 90
+    age: "60"
+```
+
+**Parameter list:**
+
+- `query` - Required. The Datadog query to run.
+- `queryValue` - Required. Value to reach to start scaling.
+- `type`- Optional. Global or Average. Whether to start scaling based on the value or the average between pods. Default: Average.
+- `metricName` - Required. Name to identify the Metric in the external.metrics.k8s.io API. If using more than one trigger it is required that all `metricName`(s) be unique.
+- `age` - Optional. The time window (in seconds) to retrieve metrics from Datadog. Default: 90. 
+
+### Authentication
+
+Datadog requires both an API key and an APP key to retrieve metrics from your account.
+
+You should use `TriggerAuthentication` CRD to configure the authentication:
+
+### Example
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: datadog-secrets
+  namespace: my-project
+type: Opaque
+data:
+  apiKey: # Required: base64 encoded value of Datadog apiKey
+  appKey: # Required: base64 encoded value of Datadog appKey
+  ddSite: # Optional: base64 encoded value of Datadog site
+---
+apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: keda-trigger-auth-datadog-secret
+  namespace: default
+spec:
+  secretTargetRef:
+    # Required: API key for your Datadog account
+  - parameter: apiKey
+    name: datadog-secrets
+    key: apiKey
+    # Required: APP key for your Datadog account
+  - parameter: appKey
+    name: datadog-secrets
+    key: appKey
+    # Optional: Datadog site. Default: "datadoghq.com"
+  - parameter: ddSite
```

Comment:
Can we rename this parameter to `datadogSite` please?
```suggestion
  - parameter: datadogSite
```

Reply:
I could change this, but DD_SITE (or ddSite) is the common name for this config option in Datadog. I think it would be more familiar to Datadog users to call it ddSite.

Reply:
We tend to avoid abbreviations so what if we:
1. Rename it to suggestion above
2. Extend documentation for this auth parameter and mention that this is commonly referred to as DD_SITE in Datadog docs?

Reply:
Sounds like a plan :)